### PR TITLE
[Bugfix] ComponentCanDeactivate Trigger 

### LINKED ISSUE #1966: Clone dialog should not activate the "ComponentCanDeactivate" dialog
In programming exercises in the exam mode, the ComponentCanDeactivate dialog is opened when students try to clone the repository.

This is confusing and should not happen in this case. We should try to prevent the dialog then.

### PR DESCRIPTION
**Pull request by anditurdiu**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
1. Clone dialog should not activate the "ComponentCanDeactivate" dialog, see #1966
2. The "ComponentCanDeactivate" dialogue was also triggered if the student is not registered to an exam, or if the student was trying to submit early.

### Description
1. Do not trigger the dialogue, if the student is handing in early or if there is no student exam
2. Fix #1966 using <iframe>

### Steps for Testing
1. Log in to Artemis
2. Go to an exam where you are not registered. 
3. Navigate away from the exam. 
=> The ComponentCanDeactivate should not trigger
4. Now go to an exam where you are registered. 
5. Try cloning a programing exercise using clone in sourcetree
=> ComponentCanDeactivate should not trigger
6. Hand in early
=> ComponentCanDeactivate should not trigger

### CODE REVIEW

File: src/main/webapp/app/exam/participate/exam-participation.component.ts [hunk 1]
```diff
@@ -159,7 +158,7 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
     // displays the alert for confirming leaving the page if there are unsaved changes
     @HostListener('window:beforeunload', ['$event'])
     unloadNotification($event: any): void {
-        if (!this.isOver()) {
+        if (!this.isOver() && this.studentExam != null && !this.handInEarly) {
```
Comment (sleiss) [inline src/main/webapp/app/exam/participate/exam-participation.component.ts:161]: I'm not sure, but can't we just use `!canDeactivate()` here? That would remove some code duplication.

Comment (anditurdiu) [reply src/main/webapp/app/exam/participate/exam-participation.component.ts:161]: yes you are right, good catch 👍

**Comment by fde312** [comment]
@anditurdiu I just found another place where canDeactivate is triggered where (I think) it is not needed. When waiting for the exam to start and then changing the screen. 
image

One could argue that this is already part of the exam, but as no changes were actually made, I think we can also remove this here :)

**Comment by anditurdiu** [comment]
> @anditurdiu I just found another place where canDeactivate is triggered where (I think) it is not needed. When waiting for the exam to start and then changing the screen.
> image
> 
> One could argue that this is already part of the exam, but as no changes were actually made, I think we can also remove this here :)

Nice catch @fde312 . I just pushed a commit where we also do not trigger the component if the exam has not been marked as started. This is set if the student confirms the exam start and the timer has run out, so the exam time has started.

**Pull request merged**